src/main/java/frc/robot/poseestimator/PoseEstimatorMath.java

```diff
@@ -0,0 +1,38 @@
+package frc.robot.poseestimator;
+
+import edu.wpi.first.math.geometry.*;
+
+public class PoseEstimatorMath {
+
+	public static double getKalmanRatio(double odometryStandardDeviation, double visionStandardDeviation) {
+		return odometryStandardDeviation / (odometryStandardDeviation + visionStandardDeviation);
+	}
+
+	public static Rotation2d combineVisionHeadingToGyro(
+		Rotation2d visionEstimatedHeading,
+		Rotation2d sampleDifferenceFromHeading,
+		Rotation2d currentEstimatedHeading,
+		double gyroStandardDeviation,
+		double visionStandardDeviation
+	) {
+		double combinedStandardDeviation = getKalmanRatio(gyroStandardDeviation, visionStandardDeviation);
+		Rotation2d estimatedHeadingAtSampleTime = currentEstimatedHeading.plus(sampleDifferenceFromHeading);
+		Rotation2d differenceFromVisionAndEstimatedHeading = getShortestAngleDifference(visionEstimatedHeading, estimatedHeadingAtSampleTime);
+		Rotation2d scaledDifferenceToAddToEstimatedHeading = differenceFromVisionAndEstimatedHeading.times(combinedStandardDeviation);
```

Comment:
This is _not_ how you use a ratio, nor stdDev. The point of a ratio is that too parts adds up to one, proportional to the ratio. This completely breaks the idea of units. 

If you trust "more", WOLOG, the vision, then you want to give more weight to the vision (e.g. using a weighted mean of the gyro difference and the vision difference from current heading), rather than moving the robot/heading more "far away" from the current heading.